modules/lm-coursier/src/main/scala/lmcoursier/CoursierDependencyResolution.scala

```diff
@@ -15,10 +16,84 @@ import sbt.util.Logger
 import coursier.core.Dependency
 import coursier.core.Publication
 
-class CoursierDependencyResolution(conf: CoursierConfiguration) extends DependencyResolutionInterface {
+import scala.util.{Try, Failure}
+
+class CoursierDependencyResolution(
+  conf: CoursierConfiguration, 
+  protocolHandlerConfiguration: Option[CoursierConfiguration],
+  bootstrapingProtocolHandler: Boolean) extends DependencyResolutionInterface {
+
+  def this(conf: CoursierConfiguration) = 
+    this(
+      conf, 
+      protocolHandlerConfiguration = None,
+      bootstrapingProtocolHandler = true
+    )
 
   lmcoursier.CoursierConfiguration.checkLegacyCache()
 
+  private var protocolHandlerClassloader: Option[ClassLoader] = None
+
+  private def fetchProtocolHandlerClassloader(
+    configuration: UpdateConfiguration,
+    uwconfig: UnresolvedWarningConfiguration,
+    log: Logger
+  ): ClassLoader = {
+
+    val conf0 = protocolHandlerConfiguration.getOrElse(conf)
+
+    def isUnknownProtocol(rawURL: String): Boolean = {
+      Try(new URL(rawURL)) match {
+        case Failure(ex) if ex.getMessage.startsWith("unknown protocol: ") => true
+        case _ => false
+      }
+    }
+
+    val confWithoutUnknownProtocol = 
+      conf0.withResolvers(
+        conf0.resolvers.filter {
+          case maven: MavenRepository => 
+            !isUnknownProtocol(maven.root)
+          case _ => 
+            true
+        }
+      )
+
+    val resolution = new CoursierDependencyResolution(
+      conf = confWithoutUnknownProtocol,
+      protocolHandlerConfiguration = None,
+      bootstrapingProtocolHandler = false
+    )
+
+    val fakeModule = 
+      ModuleDescriptorConfiguration(
+        ModuleID("lmcoursier", "lmcoursier", "0.1.0"),
+        ModuleInfo("protocol-handler")
+      )
+      .withDependencies(conf0.protocolHandlerDependencies.toVector)
+
+    val reportOrUnresolved = resolution.update(moduleDescriptor(fakeModule), configuration, uwconfig, log)
+
+    val report = reportOrUnresolved match {
+      case Right(report0) => 
+        report0
+
+      case Left(unresolvedWarning) =>
+        import sbt.util.ShowLines._
+        unresolvedWarning.lines.foreach(log.warn(_))
+        throw unresolvedWarning.resolveException
+    }
+
+    val jars = 
+      for {
+        reportConfiguration <- report.configurations.filter(_.configuration.name == "runtime")
+        module <- reportConfiguration.modules
+        (_, jar) <- module.artifacts
+      } yield jar
+
+    new URLClassLoader(jars.reverse.map(_.toURI().toURL()).toArray)
```

Comment:
Is there a particular reason to reverse `jars` here?

Reply:
Hum, this looks like old code from an issue related to classloading I was debugging. Reverting.